From 53c51e09ef68debd43a1977b2c2c31ba9e290f9a Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Sat, 25 Nov 2023 19:09:53 +0100
Subject: [PATCH] Set libkonsoleprivate and libkonsoleapp version to release
 version

It's a private library, so its ABI may change any release.

This also makes konsole6-part co-installable with konsole5-part.
---
 CMakeLists.txt     | 3 ---
 src/CMakeLists.txt | 6 ++----
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 35aacbafa..3f2d0507e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,9 +43,6 @@ include(CheckIncludeFiles)
 # Allows passing e.g. -DECM_ENABLE_SANITIZERS='address;undefined' to cmake.
 include(ECMEnableSanitizers)
 
-ecm_setup_version(${RELEASE_SERVICE_VERSION} VARIABLE_PREFIX KONSOLEPRIVATE
-    SOVERSION 1
-)
 find_package(Qt6 ${QT_MIN_VERSION} CONFIG REQUIRED
     Core
     DBus
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7ecfd4435..64c7aa916 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -288,8 +288,7 @@ target_link_libraries(konsoleprivate
 )
 
 set_target_properties(konsoleprivate PROPERTIES
-    VERSION ${KONSOLEPRIVATE_VERSION}
-    SOVERSION ${KONSOLEPRIVATE_SOVERSION}
+    VERSION ${RELEASE_SERVICE_VERSION}
 )
 
 install(TARGETS konsoleprivate ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
@@ -325,8 +324,7 @@ target_link_libraries(konsoleapp
 )
 
 set_target_properties(konsoleapp PROPERTIES
-    VERSION ${KONSOLEPRIVATE_VERSION}
-    SOVERSION ${KONSOLEPRIVATE_SOVERSION}
+    VERSION ${RELEASE_SERVICE_VERSION}
 )
 
 install(TARGETS konsoleapp ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
-- 
GitLab

